Matlab/Matlab.sublime-syntax

```diff
@@ -48,477 +76,792 @@ contexts:
+              scope: keyword.operator.properties.matlab
+            - include: argument-placeholder
+            - include: builtin-types
+            - include: expressions
+    - match: (?=\S)
+      set: function-body
+
+  function-body:
+    - meta_scope: meta.function.matlab
+    - match: \bend\b
+      scope: keyword.control.end.function.matlab
+      pop: true
+    - include: function-declaration
+    - include: keywords
+    - include: expressions
+
+  class-declaration:
+    - match: \bclassdef\b
+      scope: keyword.declaration.class.matlab
       push:
-        - meta_scope: meta.classdef.matlab
+        - meta_scope: meta.class.matlab
+        - include: eol-pop
         - match: \(
-          scope: punctuation.definition.properties.begin.matlab
+          scope: punctuation.section.parens.begin.matlab
           push:
-            - meta_scope: meta.properties.matlab
+            - meta_scope: meta.attributes.matlab
+            - include: line-continuation
+            - include: eol-pop
             - match: \)
-              scope: punctuation.definition.properties.end.matlab
+              scope: punctuation.section.parens.end.matlab
               pop: true
-            - match: ','
-              scope: punctuation.separator.matlab
-            - match: \b(Abstract|AllowedSubclasses|ConstructOnLoad|HandleCompatible|Hidden|InferiorClasses|Sealed)\s*(=)
-              captures:
-                1: variable.parameter.matlab
-                2: keyword.operator.symbols.matlab
-            - match: \b(false|true)\b
-              scope: constant.language.matlab
-        - match : \b(\w+)\b(?:\s*(<)\s*(\w+))?
-          captures:
-            1: entity.name.class.matlab
-            2: punctuation.definition.inheritance.matlab
-            3: entity.other.inherited-class.matlab
+            # https://www.mathworks.com/help/matlab/matlab_oop/class-attributes.html
+            - match: \b(?:Abstract|AllowedSubclasses|ConstructOnLoad|HandleCompatible|Hidden|InferiorClasses|Sealed)\b
+              scope: variable.parameter.attribute.matlab
+            - include: expressions
+        - match: '{{identifier}}'
+          scope: entity.name.class.matlab
+          set:
+            - meta_content_scope: meta.class.matlab
+            - include: line-continuation
+            - match: '{{eol}}'
+              set: class-body
+            - match: <
+              scope: punctuation.separator.inheritance.matlab
+              set:
+                - meta_content_scope: meta.class.matlab
+                - include: line-continuation
+                - match: '{{eol}}'
+                  set: class-body
+                - match: '{{identifier}}'
+                  scope: entity.other.inherited-class.matlab
+                - match: \&
+                  scope: punctuation.separator.sequence.matlab
+
+  class-body:
+    - meta_scope: meta.class.matlab
+    - match: \bend\b
+      scope: keyword.control.end.class.matlab
```

Comment:
I think I first scoped them with `keyword.control.end.<statement>.matlab`, because the single rule for `end` which was used before in this syntax assigned `keyword.control`, and with this it would still be easy for color schemes to target all `end` keywords with a single scope. But after thinking a bit about your argument, I agree that it probably is an advantage to have it scoped similar to the "opening" keyword.

I went with your second option now, i.e. `keyword.declaration.class.end`, because it is consistent with `keyword.control.{loop|conditional}.end` to have the "end" part of the scope at the 4th level, and because if a color scheme wants to use a special color for all `end` keywords, it needs to target more than only a single scope anyway now.